moto/awslambda/models.py

```diff
@@ -735,7 +741,12 @@ def update_from_cloudformation_json(
     def _create_zipfile_from_plaintext_code(code):
         zip_output = io.BytesIO()
         zip_file = zipfile.ZipFile(zip_output, "w", zipfile.ZIP_DEFLATED)
-        zip_file.writestr("lambda_function.zip", code)
+        zip_file.writestr("index.py", code)
```

Comment:
Existing bug, exposed in `test_cloudformation_stack_integration.py::test_lambda_function`.
The test created a Lambda-function, and provided the code as part of the CF template. Execution of this function was not possible before this change though, as Lambda looks for source code in a file called `index.py` by default.